src/api/models/requests.model.js

```diff
@@ -0,0 +1,141 @@
+const mongoose = require('mongoose');
+const httpStatus = require('http-status');
+const { isEmpty, compact, omitBy, isNil } = require('lodash');
+const bcrypt = require('bcryptjs');
+const moment = require('moment-timezone');
+const jwt = require('jwt-simple');
+const uuidv4 = require('uuid/v4');
+const APIError = require('../utils/APIError');
+const ResetToken = require('./resetToken.model');
+const RefreshToken = require('./refreshToken.model');
+const { env, jwtSecret, jwtExpirationInterval } = require('../../config/vars');
+
+/**
+ * User Roles
+ */
+const roles = ['user', 'admin'];
+
+/**
+ * Request Schema - for logging account requests
+ * @private
+ */
+const requestSchema = new mongoose.Schema(
+  {
+    email: {
+      type: String,
+      match: /^\S+@\S+\.\S+$/,
+      required: true,
+      trim: true,
+      lowercase: true,
+    },
+    first_name: {
+      type: String,
+      required: true,
+      trim: true,
+      minlength: 1,
+    },
+    last_name: {
+      type: String,
+      required: true,
+      trim: true,
+      minlength: 1,
+    },
+    eos_account: {
+      type: String,
+      match: /^[.12345a-z]+$/,
+      required: true,
+      unique: true,
+      trim: true,
+      lowercase: true,
+      minlength: 1,
+      maxlength: 13,
+    },
+    owner_key: {
+      type: String,
+      required: true,
+      length: 64,
+    },
+    active_key: {
+      type: String,
+      required: true,
+      length: 64,
+    },
+  },
+  {
+    timestamps: true,
+    collection: 'Requests',
+    toJSON: { virtuals: true },
+  },
+);
+
+/**
+ * Add your
+ * - pre-save hooks
+ * - validations
+ * - virtuals
+ */
+
+/**
+ * Methods
+ */
+requestSchema.method({});
+
+/**
+ * Statics
+ */
+requestSchema.statics = {
+  roles,
+
+  /**
+   * List Requests in descending order of 'createdAt' timestamp.
+   *
+   * @param {number} skip - Number of entities to be skipped.
+   * @param {number} limit - Limit number of entities to be returned.
+   * @param {Object} [sort] - Object w/ keys matching fieldnames to be sorted, values as -1 (desc), 1 (asc)
+   * @param {Object} [filter] - Object matching a Mongoose query object
+   * @param {Object|String} [projection] - Mongoose `select()` arg denoting fields to include or exclude
+   * @returns {Promise<User[]>}
+   */
+  list({ skip = 0, limit = 30, sort, filter, projection }) {
+    const $match = isEmpty(filter) ? null : { $match: filter };
+    const $project = projection ? { $project: projection } : null;
+    const $skip = { $skip: skip };
+    const $limit = { $limit: limit };
+    const $sort = sort ? { $sort: sort } : null;
+
+    const agg = compact([$match, $project, $sort, $skip, $limit]);
+
+    return this.aggregate(agg).exec();
+  },
+
+  /**
+   * Return new validation error
+   * if error is a mongoose duplicate key error
+   *
+   * @param {Error} error
+   * @returns {Error|APIError}
+   */
+  checkDuplicateEmail(error) {
```

Comment:
only one EOS account per email allowed?

Reply:
Yeah, right now I don't have any qualifications for that.  So that's what's in there right now